src/create_wand_data.cpp

```diff
@@ -9,53 +10,68 @@
 #include "util/util.hpp"
 #include "wand_data.hpp"
 #include "wand_data_compressed.hpp"
-#include "wand_data_raw.hpp"
 #include "wand_data_range.hpp"
+#include "wand_data_raw.hpp"
 
 #include "CLI/CLI.hpp"
 
-int main(int argc, const char **argv) {
+int main(int argc, const char **argv)
+{
     using namespace pisa;
 
     float lambda = 0.0f;
+    uint64_t fixed_block_size = 0;
     std::string input_basename;
     std::string output_filename;
-    bool        variable_block = false;
-    bool        compress       = false;
-    bool        range          = false;
+    bool variable_block = false;
+    bool compress = false;
+    bool range = false;
 
     CLI::App app{"create_wand_data - a tool for creating additional data for query processing."};
     app.add_option("-c,--collection", input_basename, "Collection basename")->required();
     app.add_option("-o,--output", output_filename, "Output filename")->required();
-    app.add_option("-l,--lambda", lambda, "Lambda parameter for variable blocks");
     auto var_block_opt = app.add_flag("--variable-block", variable_block, "Variable length blocks");
+    auto var_block_param_opt =
+        app.add_option("-b,--block-size", fixed_block_size, "Block size for fixed-length blocks")
+            ->excludes(var_block_opt);
+    app.add_option("-l,--lambda", lambda, "Lambda parameter for variable blocks")
+        ->excludes(var_block_param_opt)
+        ->needs(var_block_opt);
     app.add_flag("--compress", compress, "Compress additional data");
     app.add_flag("--range", range, "Create docid-range based data")->excludes(var_block_opt);
 
-    
     CLI11_PARSE(app, argc, argv);
 
-    partition_type p_type =
-        variable_block ? partition_type::variable_blocks : partition_type::fixed_blocks;
-
-    std::string partition_type_name =
-        (p_type == partition_type::fixed_blocks) ? "static partition" : "variable partition";
+    std::string partition_type_name = (lambda) ? "variable partition" : "static partition";
     spdlog::info("Block based wand creation with {}", partition_type_name);
 
-    binary_collection      sizes_coll((input_basename + ".sizes").c_str());
+    binary_collection sizes_coll((input_basename + ".sizes").c_str());
     binary_freq_collection coll(input_basename.c_str());
 
+    // Initialize the variant to the correct type, default arguments from config
```

Comment:
How about we use the following code here. For one, it makes the block size `const`, and then also is clear. Rule of thumb: if you have to describe a specific block/branch in a comment, you most likely want to rearrange things to make it more clear---and then remove the comments.

```cpp
auto const block_size = [&]() -> BlockSize {
    if (variable_block) {
        return fixed_block_size ? FixedBlock(*fixed_block_size) : FixedBlock();
    } else {
        return lambda ? VariableBlock(*lambda) : VariableBlock();
    }
}();
```

Reply:
This is elegant, thanks! Incorporating it now.